app/src/main/java/ai/elimu/appstore/BaseApplication.java

```diff
@@ -72,6 +78,16 @@ protected void log(int priority, String tag, String message, Throwable throwable
         Database db = helper.getWritableDb();
         daoSession = new CustomDaoMaster(db).newSession();
 
+        int oldVersionCode = AppPrefs.getAppVersionCode();
+        if (oldVersionCode < 2000010) {
```

Comment:
This block of code should only trigger if the application is being upgraded from a previously installed version. Now it also triggers if the app is started for the very first time. Can this be changed?

Reply:
Sure, then I'll move the deletion logic to onUpgrade (CustomDaoMaster.java) in case oldVersion < 2000010.

Reply:
It's only triggered when there is a database version upgrade.
Normal app versionCode upgrade does not relates.

Reply:
Ok. Those two cases should be handled separately.